agent/command/s3_put.go

```diff
@@ -96,17 +97,19 @@ type s3put struct {
 	// PatchOnly defaults to false. If set to true, this command will noop without error for non-patch tasks.
 	PatchOnly string `mapstructure:"patch_only" plugin:"patch_only"`
 
-	// SkipExisting, when set to true, will not upload files if they already exist in s3.
+	// SkipExisting, when set to 'true', will not upload files if they already
+	// exist in s3. This will not cause the s3.put command to fail. When set to
+	// 'error' instead, the command will fail if a remote file already exists.
+	// This beavior respects s3's strong read-after-write consistency model.
 	SkipExisting string `mapstructure:"skip_existing" plugin:"expand"`
```

Comment:
I think skip_existing: error doesn't actually make a ton of logical sense -- we aren't skipping an existing file (i.e. succeeding), we're choosing to fail. Parameters are free so I think a new parameter like `error_on_existing: true` is more clear. This also maintains our behavior of parsing booleans.

Reply:
After thinking this over I agree with you here and also think this doesn't make sense as a feature at all. I reverted back to the old behavior where this is just toggled as a plain boolean and there is no option to error out.